# Hugging Face

Hugging Face is an American company that develops computation tools for building applications using machine learning. It is most notable for its transformers library built for natural language processing applications and its platform that allows users to share machine learning models and datasets and showcase their work. 

The goal of this tutorial is to introduce how to use dataset from hugging face.

## Searching for datasets

First go to the `datasets` tab of the Hugging face website (https://huggingface.co/datasets) and search for what dataset your interesting of. 

For this tutorials, the goal is to use a dataset that contain images of mathematical formula generated thanks to latex code and there latex equivalent. For example, there is this dataset: https://huggingface.co/datasets/OleehyO/latex-formulas

On this page, you can see a preview of the dataset, some informations about the dataset (size:2.78GB, number of formulas(rows):1,558,585, ...) and a quick utilization instructions sometimes.

## Load dataset

First, we will use the `datasets` Python package and load the dataset (you have to have access to a internet connection to load the dataset).

In [19]:
from datasets import load_dataset
ds = load_dataset("OleehyO/latex-formulas", "cleaned_formulas")

Note that, here the `"cleaned_formulas"` option is to load a specific dataset. To see the details, look forward on the website page.

## Using dataset

In [8]:
print(type(ds))
print(ds)

<class 'datasets.dataset_dict.DatasetDict'>
DatasetDict({
    train: Dataset({
        features: ['image', 'latex_formula'],
        num_rows: 552340
    })
})


Has you can see, here our dataset only contain one set: `train`. Hovewer, lots of datasets contains multiple sets like: `train`, `validation` and `test`.

Moreover, the `train` dataset contain two keys: `image` and `latex_formula` and have 552340 rows.

To access any rows you can do:

In [17]:
train = ds['train']
row_0 = train[0] #The first row of the dataset
print(row_0['image'])
print(row_0['latex_formula'])

<PIL.JpegImagePlugin.JpegImageFile image mode=RGB size=323x44 at 0x760E1ACBBD00>
\begin{align*}-\div (y^{1-2m}\nabla w)=0\mbox{in}\mathbb R^n\times\mathbb R_+;w\big|_{y=0}=|u|\end{align*}
